stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -371,64 +371,92 @@ class connection:
     ProgrammingError: Any
     Warning: Any
     @property
-    def async_(self) -> Any: ...
+    def async_(self) -> int: ...
     autocommit: Any
     @property
     def binary_types(self) -> Any: ...
     @property
-    def closed(self) -> Any: ...
+    def closed(self) -> int: ...
     cursor_factory: Callable[..., _cursor]
-    deferrable: Any
     @property
-    def dsn(self) -> Any: ...
+    def dsn(self) -> str: ...
     @property
-    def encoding(self) -> Any: ...
+    def encoding(self) -> str: ...
     @property
     def info(self) -> ConnectionInfo: ...
     isolation_level: Any
     notices: Any
     notifies: Any
     @property
-    def pgconn_ptr(self) -> Any: ...
+    def pgconn_ptr(self) -> int | None: ...
     @property
     def protocol_version(self) -> int: ...
+    # For these two the getter returns bool | None, but the setter
```

Comment:
Could also consider using read-write properties that return different types from their getter to the ones they receive in their setter, like we do here:

https://github.com/python/typeshed/blob/62a8a6922cb052704198ebd28d26887d3b6d17f4/stdlib/builtins.pyi#L82

Reply:
Oh good point, I thought mypy didn't support that.